src/projects/dualBackedToOne/MDualBackedToOne.sol

```diff
@@ -0,0 +1,294 @@
+    }
+
+    /// @inheritdoc IERC20
+    function totalSupply() public view override returns (uint256) {
+        return _getMDualBackedToOneStorageLocation().totalSupply;
+    }
+
+    /// @inheritdoc IMDualBackedToOne
+    function secondaryBacker() public view returns (IERC20) {
+        return _getMDualBackedToOneStorageLocation().secondaryBacker;
+    }
+
+    /// @inheritdoc IMDualBackedToOne
+    function secondarySupply() public view returns (uint256) {
+        return _getMDualBackedToOneStorageLocation().secondarySupply;
+    }
+
+    /// @inheritdoc IMDualBackedToOne
+    function yield() public view returns (uint256) {
+        unchecked {
+            uint256 balance_ = _mBalanceOf(address(this));
+            uint256 mBacking_ = totalSupply() - secondarySupply();
+
+            return balance_ > mBacking_ ? balance_ - mBacking_ : 0;
+        }
+    }
+
+    /// @inheritdoc IMDualBackedToOne
+    function claimYield() public returns (uint256) {
+        _beforeClaimYield();
+
+        uint256 yield_ = yield();
+
+        if (yield_ == 0) return 0;
+
+        emit YieldClaimed(yield_);
+
+        _mint(yieldRecipient(), yield_);
+
+        return yield_;
+    }
+
+    function setYieldRecipient(address account) external onlyRole(YIELD_RECIPIENT_MANAGER_ROLE) {
+        claimYield();
+
+        _setYieldRecipient(account);
+    }
+
+    function yieldRecipient() public view returns (address) {
+        return _getMDualBackedToOneStorageLocation().yieldRecipient;
+    }
+
+    function wrapSecondary(address recipient, uint256 amount) external onlySwapFacility {
+        // NOTE: `msg.sender` is always SwapFacility contract.
+        //       `ISwapFacility.msgSender()` is used to ensure that the original caller is passed to `_beforeWrap`.
+        _wrapSecondary(ISwapFacility(msg.sender).msgSender(), recipient, amount);
+    }
+
+    function replaceSecondary(uint256 amount) external onlyRole(COLLATERAL_MANAGER_ROLE) {
+        IMTokenLike(mToken).transferFrom(msg.sender, address(this), amount);
+
+        MDualBackedToOneStorageStruct storage $ = _getMDualBackedToOneStorageLocation();
+
+        $.secondaryBacker.transfer(msg.sender, _scaleDecimals(amount));
+
+        $.secondarySupply -= amount;
+
+        emit SecondaryBackingReplaced(amount);
+    }
+
+    function _wrapSecondary(address account, address recipient, uint256 amount) internal {
+        _revertIfInvalidRecipient(recipient);
+        _revertIfInsufficientAmount(amount);
+
+        // NOTE: Invoke the _beforeUnwrap() hook from yield to one
+        _beforeWrap(account, recipient, amount);
+
+        // NOTE: `msg.sender` is always SwapFacility contract.
+        // NOTE: The behavior of `IMTokenLike.transferFrom` is known, so its return can be ignored.
+        MDualBackedToOneStorageStruct storage $ = _getMDualBackedToOneStorageLocation();
+
+        $.secondaryBacker.transferFrom(msg.sender, address(this), _scaleDecimals(amount));
+
+        unchecked {
```

Comment:
unchecked  blocks should be consistent through the codebase

Reply:
Has been fixed in a previous commit.